src/lib_ccaligner/params.cpp

```diff
@@ -6,55 +6,35 @@
 
 #include "params.h"
 
-const std::string currentTime()
-{
-    time_t now = time(0);
-    struct tm tstruct = * localtime(&now);
-    char localTime[32]; //being generous
-    strftime(localTime, sizeof(localTime), "%d-%m-%Y-%H-%M-%S", &tstruct);
-    return std::string(localTime);
+#include <sstream>
+#include <chrono>
+#include <iomanip>
+
+// Default paths.
+namespace {
+    constexpr auto defaultModelPath = "model/";
+    constexpr auto defaultLmPath = "tempFiles/lm/complete.lm";
+    constexpr auto defaultDictPath = "tempFiles/dict/complete.dict";
+    constexpr auto defaultFsgPath = "tempFiles/fsg/";
+    constexpr auto defaultPhoneticLmPath = "model/en-us-phone.lm.bin";
 }
 
 
-Params::Params()
+Params::Params() noexcept
+    : modelPath(defaultModelPath), lmPath(defaultLmPath), dictPath(defaultDictPath), fsgPath(defaultFsgPath), phoneticlmPath(defaultPhoneticLmPath),
+      sampleWindow(), audioWindow(), searchWindow(3), chosenAlignerType(asrAligner), grammarType(complete_grammar), outputFormat(xml),
+      printOption(printBothWithDistinctColors), verbosity(true), useFSG(), transcribe(), useBatchMode(), useExperimentalParams(),
+      searchPhonemes(), displayRecognised(true), readStream(), quickDict(), quickLM(), audioIsRaw()
```

Comment:
I think it would be more clear if you put them in the header file

Reply:
@harrynull You mean initialize them in the class definition instead? Good idea! Thanks :)

Reply:
@harrynull Actually, that would mean defining each variable in its own line (the lines would get too crowded otherwise). Is that still ok?